File: src/react_app/src/interface/admin/Sponsor.ts
export interface SponsorTier {
  code: string;
  name: string;
  sortSeq?: number | null;
}

export interface SponsorListItem {
  sponsorSeq: number;
  editionYear: number;
  tierCd: string;
  tierCdNm?: string | null;
  tierSortSeq?: number | null;
  companyName: string;
  logoFileSeq?: number | null;
  logoFileDtlSeq?: number | null;
  logoFilePath?: string | null;
  logoFileUrl?: string | null;
  hasRemarks?: boolean | null;
  description?: string | null;
  representativeRemarks?: string | null;
  websiteUrl?: string | null;
  sortSeq: number;
  useYn: string;
  rgstUserName?: string | null;
  rgstDateTime?: string | null;
  uptUserName?: string | null;
  uptDateTime?: string | null;
}

export interface SponsorDetail extends SponsorListItem {
  rgstUserSeq?: number | null;
  uptUserSeq?: number | null;
}

export interface SponsorSavePayload {
  editionYear: number;
  tierCd: string;
  companyName: string;
  logoFileSeq?: number | null;
  description?: string | null;
  representativeRemarks?: string | null;
  websiteUrl?: string | null;
  sortSeq: number;
  useYn: string;
}

export interface SponsorSearchParams {
  editionYear?: number;
  tierCd?: string;
  useYn?: string;
  keyword?: string;
}

File: src/main/java/com/kcabEvent/domain/Sponsor.java
package com.kcabEvent.domain;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
public class Sponsor {
    private Long sponsorSeq;
    /** 년도별 선발 */
    private Integer editionYear;
    /** SPONSOR_TIER 공통코드 (event.com_cd.com_cd) */
    private String tierCd;
    private String companyName;
    /** 기업 로고 파일 그룹 (event.tb_file.file_seq) */
    private Long logoFileSeq;
    /** 기업설명 (HTML) */
    private String description;
    /** Representative's Remarks (HTML, 선택) */
    private String representativeRemarks;
    private Integer sortSeq;
    private String useYn;
    /** DB: event.sponsor.created_by */
    private Long rgstUserSeq;
    /** DB: event.sponsor.created_at */
    private LocalDateTime rgstDateTime;
    /** DB: event.sponsor.updated_by */
    private Long uptUserSeq;
    /** DB: event.sponsor.updated_at */
    private LocalDateTime uptDateTime;
}
